### ISSUE TITLE
Variations REST API is not returning options with the number zero (0)

### ISSUE THREAD

**Issue by rsaftoiu** [woocommerce/woocommerce#21743]
**Describe the bug**
A customer creates a variable product in WooCommerce, where one of the options is the number zero (0). When the variations of this variable product are queried via the rest api, the attributes for this variant do not get returned.

**To Reproduce**
1. Go to the Products Page
2. Create a new variable product with options 6 | 0 | 2, as seen here: 
3. Generate all variations, as seen here: 
4. Let's say the product_id is 193. Query the product api: /wp-json/wc/v2/products/193
5. The attributes in the response should look like this: 
 "attributes": [
        {
            "id": 0,
            "options": [
                "6",
                "0",
                "2"
            ],
            "name": "Options",
            "variation": true,
            "visible": true,
            "position": 0
        }
    ],

6. Now, query the variant that got assigned the option 0. Let's assume its id was 196: /wp-json/wc/v2/products/193/variations/196
7. Observe the attributes come back as an empty array [].
8. Observe that for all other variants the attributes are correct.

**Screenshots**
Item Attributes: ![variants](https://user-images.githubusercontent.com/4667234/47674121-cce3c000-db73-11e8-9e27-64025eb2fac6.png)
Variants: ![item_attributes](https://user-images.githubusercontent.com/4667234/47674122-cd7c5680-db73-11e8-892e-6aeb2dae0e03.png)
Item response json: https://drive.google.com/file/d/1ran1ihffnO-JxJoTeUnMOqomiWdej86H/view?usp=sharing
Variations response json: https://drive.google.com/file/d/1UMgPhNil6nIPDuN8l91GSpLGciZlYqma/view?usp=sharing

**Expected behavior**
I expect the attributes to be returned for the variant with the 0 option.

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

### WordPress Environment ###

Home URL: https://mywoostore.net
Site URL: https://mywoostore.net
WC Version: 3.5.0
Log Directory Writable: ✔
WP Version: 4.9.8
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache/2.4.18 (Ubuntu)
PHP Version: 7.0.32-0ubuntu0.16.04.1 - We recommend using PHP version 7.2 or above for greater performance and security. How to update your PHP version
PHP Post Max Size: 16 MB
PHP Time Limit: 60
PHP Max Input Vars: 1000
cURL Version: 7.47.0
OpenSSL/1.0.2g

SUHOSIN Installed: –
MySQL Version: 5.7.24-0ubuntu0.16.04.1
Max Upload Size: 16 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.5.0
WC Database Prefix: wp_
MaxMind GeoIP Database: ✔
Total Database Size: 3.32MB
Database Data Size: 2.03MB
Database Index Size: 1.29MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.05MB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB
wp_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB
wp_comments: Data: 0.02MB + Index: 0.09MB
wp_failed_jobs: Data: 0.02MB + Index: 0.00MB
wp_links: Data: 0.02MB + Index: 0.02MB
wp_mailchimp_carts: Data: 0.02MB + Index: 0.00MB
wp_options: Data: 1.03MB + Index: 0.05MB
wp_postmeta: Data: 0.36MB + Index: 0.44MB
wp_posts: Data: 0.08MB + Index: 0.06MB
wp_queue: Data: 0.02MB + Index: 0.00MB
wp_termmeta: Data: 0.02MB + Index: 0.03MB
wp_terms: Data: 0.02MB + Index: 0.03MB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB
wp_usermeta: Data: 0.02MB + Index: 0.03MB
wp_users: Data: 0.02MB + Index: 0.05MB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB

### Post Type Counts ###

attachment: 4
page: 5
post: 2
product: 28
product_variation: 129
scheduled-action: 13
shop_order: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (1) ###

WooCommerce: by Automattic – 3.5.0

### Settings ###

API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

### WC Pages ###

Shop base: #4 - /shop/
Cart: #5 - /cart/
Checkout: #6 - /checkout/
My account: #7 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 2.3.2 – 2.3.5 is available
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

**Comment by rrennick** [comment]
Have you tested with `v3` of the REST api? ex `/wp-json/wc/v3/products/125/variations/127`

**Comment by zfwjs** [comment]
Same result.

11728 is the parent product, 11729-11736 are all variations of the parent (Attribute name Size, values 00 | 0 | 2 | 4 | 6 | 8 | 10 | 12).

Querying the 00 variation using:

/wp-json/wc/v3/products/11729
or
/wp-json/wc/v3/products/11728/variations/11729

returns:

  "attributes": [
    {
      "id": 0,
      "name": "Size",
      "option": "00"
    },

Querying the 0 variation using:

/wp-json/wc/v3/products/11730
or
/wp-json/wc/v3/products/11728/variations/11730

returns:

  "attributes": [],

**Comment by rrennick** [comment]
I created a new product to test with `Size` : `00 | 0 | 2 | 4`, used `https://ron.localhost/wp-json/wc/v3/products/127/variations`

```json
 [
  {
    "id": 131,
    "date_created": "2018-11-23T13:13:10",
    "date_created_gmt": "2018-11-23T13:13:10",
    "date_modified": "2018-11-23T13:13:10",
    "date_modified_gmt": "2018-11-23T13:13:10",
    "description": "",
    "permalink": "https://ron.localhost/product/variable-attr-00/?attribute_size=4",
    "attributes": [
      {
        "id": 0,
        "name": "Size",
        "option": "4"
      }
    ],
    "menu_order": 4,
    "_links": {
      "self": [
        {
          "href": "https://ron.localhost/wp-json/wc/v3/products/127/variations/131"
        }
      ],
    }
  },
  {
    "id": 130,
    "date_created": "2018-11-23T13:13:10",
    "date_created_gmt": "2018-11-23T13:13:10",
    "date_modified": "2018-11-23T13:13:10",
    "date_modified_gmt": "2018-11-23T13:13:10",
    "description": "",
    "permalink": "https://ron.localhost/product/variable-attr-00/?attribute_size=2",
    "attributes": [
      {
        "id": 0,
        "name": "Size",
        "option": "2"
      }
    ],
    "menu_order": 3,
    "_links": {
      "self": [
        {
          "href": "https://ron.localhost/wp-json/wc/v3/products/127/variations/130"
        }
      ]
    }
  },
  {
    "id": 129,
    "date_created": "2018-11-23T13:13:10",
    "date_created_gmt": "2018-11-23T13:13:10",
    "date_modified": "2018-11-23T13:13:10",
    "date_modified_gmt": "2018-11-23T13:13:10",
    "description": "",
    "permalink": "https://ron.localhost/product/variable-attr-00/",
    "attributes": [],
    }
  },
  {
    "id": 128,
    "date_created": "2018-11-23T13:13:10",
    "date_created_gmt": "2018-11-23T13:13:10",
    "date_modified": "2018-11-23T13:13:10",
    "date_modified_gmt": "2018-11-23T13:13:10",
    "description": "",
    "permalink": "https://ron.localhost/product/variable-attr-00/?attribute_size=00",
    "attributes": [
      {
        "id": 0,
        "name": "Size",
        "option": "00"
      }
    ],
    "menu_order": 1,
    "_links": {
      "self": [
        {
          "href": "https://ron.localhost/wp-json/wc/v3/products/127/variations/128"
        }
      ],
    }
  }
]
```

Confirmed that the attributes are empty in the `0` size of the variations collection. They are also empty on `https://ron.localhost/wp-json/wc/v3/products/127/variations/129`. 

A second issue is that the permalink for the size `0` variation does not go directly to that variation.

**Comment by claudiulodro** [comment]
Should be fixed in #22026
